diff --git a/lib/spritesheet/Spritesheet.d.ts b/lib/spritesheet/Spritesheet.d.ts
index 8442257e955d251a01d88631e3958ef39bc3b3c4..a37b32440814b531b1d23a7e0afbe4151eba2ad8 100644
--- a/lib/spritesheet/Spritesheet.d.ts
+++ b/lib/spritesheet/Spritesheet.d.ts
@@ -270,10 +270,15 @@ export declare class Spritesheet<S extends SpritesheetData = SpritesheetData> {
      */
     constructor(texture: BindableTexture, data: S);
     /**
-     * Parser spritesheet from loaded data. This is done asynchronously
+     * Parse spritesheet from loaded data. This is done asynchronously
      * to prevent creating too many Texture within a single process.
      */
     parse(): Promise<Record<string, Texture>>;
+    /**
+     * Parse spritesheet from loaded data synchronously.
+     * All frames are processed immediately.
+     */
+    parseSync(): Record<keyof S['frames'], Texture>;
     /**
      * Process a batch of frames
      * @param initialFrameIndex - The index of frame to start.
diff --git a/lib/spritesheet/Spritesheet.mjs b/lib/spritesheet/Spritesheet.mjs
index 799446b8bc5fe9ea2d6eecbbc195a70fec52530a..a26071311a998d249e133cd4c34e92492dcf268e 100644
--- a/lib/spritesheet/Spritesheet.mjs
+++ b/lib/spritesheet/Spritesheet.mjs
@@ -50,13 +50,26 @@ const _Spritesheet = class _Spritesheet {
       }
     });
   }
+  /**
+   * patched in from https://github.com/pixijs/pixijs/pull/11794 until the mainline stable pixi.js release
+   * includes this
+   * 
+   * Parse spritesheet from loaded data synchronously.
+   * All frames are processed immediately.
+   */
+  parseSync() {
+    this._processFrames(0, true);
+    this._processAnimations();
+    return this.textures;
+  }
   /**
    * Process a batch of frames
    * @param initialFrameIndex - The index of frame to start.
+   * @param processAll - Whether to process all frames at once.
    */
-  _processFrames(initialFrameIndex) {
+  _processFrames(initialFrameIndex, processAll = false) {
     let frameIndex = initialFrameIndex;
-    const maxFrames = _Spritesheet.BATCH_SIZE;
+    const maxFrames = processAll ? Infinity : _Spritesheet.BATCH_SIZE;
     while (frameIndex - initialFrameIndex < maxFrames && frameIndex < this._frameKeys.length) {
       const i = this._frameKeys[frameIndex];
       const data = this._frames[i];
